scripts/skills/perks/perk_rf_target_practice.nut

```diff
@@ -16,40 +16,38 @@ this.perk_rf_target_practice <- ::inherit("scripts/skills/skill", {
 		this.m.ItemActionOrder = ::Const.ItemActionOrder.First;
 	}
 
+	function isEnabled()
+	{
+		if (this.getContainer().getActor().isDisarmed())
+			return false;
+
+		local weapon = this.getContainer().getActor().getMainhandItem();
+		return weapon != null && weapon.isWeaponType(::Const.Items.WeaponType.Bow);
+	}
+
 	function onAnySkillUsed( _skill, _targetEntity, _properties )
 	{
-		if (!_skill.isRanged() || _targetEntity == null) return;
+		if (_targetEntity == null || !_skill.isRanged() || !_skill.m.IsWeaponSkill || !this.isEnabled())
+			return;
 
-		if (_targetEntity.isArmedWithRangedWeapon() || ::Tactical.Entities.getAlliedActors(_targetEntity.getFaction(), _targetEntity.getTile(), 1, true).len() == 0)
+		if (_targetEntity.isArmedWithRangedWeapon() || ::Const.Tactical.Common.getBlockedTiles(this.getContainer().getActor().getTile(), _targetEntity.getTile(), this.getContainer().getActor().getFaction()).len() == 0);
 		{
 			_properties.RangedSkill += this.m.RangedSkillBonus;
 		}
 	}
 
 	function onGetHitFactors( _skill, _targetTile, _tooltip )
 	{
+		if (!_skill.isRanged() || !_skill.m.IsWeaponSkill || !this.isEnabled())
+			return;
+
 		local targetEntity = _targetTile.getEntity();
-		if (_skill.isRanged() && targetEntity != null && (targetEntity.isArmedWithRangedWeapon() || ::Tactical.Entities.getAlliedActors(targetEntity.getFaction(), targetEntity.getTile(), 1, true).len() == 0))
+		if (targetEntity != null && (targetEntity.isArmedWithRangedWeapon() || ::Const.Tactical.Common.getBlockedTiles(this.getContainer().getActor().getTile(), targetEntity.getTile(), this.getContainer().getActor().getFaction()).len() == 0))
 		{
 			_tooltip.push({
 				icon = "ui/tooltips/positive.png",
 				text = "[color=" + ::Const.UI.Color.PositiveValue + "]" + this.m.RangedSkillBonus + "%[/color] " + this.getName()
```

Comment:
Might aswell improve the tooltip coloring if you are already working on this perk
`text = ::MSU.Text.colorGreen(this.m.RangedSkillBonus + "%") + this.getName()`

Reply:
I avoided doing these changes in general in this perk rework so as not to muddle the actual perk changes with extra lines of just refactoring the tooltip. I'd rather do these changes separately in a single commit in the entire codebase.